### PR TITLE
feat: make rendering testing work with enigma-mocker and test fixtures 

### PR DESCRIPTION
**Pull request by LiKang6688**
Nebula serve using mocked Qlik core engine (Enigma mocker) to render the visualization charts based on the testing fixture file

Testing cases using Puppeteer capture screenshots of those rendered charts and matches against the baseline to ensure our charts look as intended and present clients with expected visuals.

Visual regression Testing flow:

Start Nebula serve and launch Puppeteer
Iterate testing fixture files
Create test case per testing fixture file
Render chart based on testing fixture file in Nebula serve using Enigma mocker
Puppeteer capture screenshot
Compare screenshot with the baseline image

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
   build:
     <<: *defaults
     docker:
-      - image: circleci/node:14.15.0
+      - image: circleci/node:16.7-browsers
```
Comment (haxxmaxx) [inline .circleci/config.yml:49]: Maybe we should bump in package too?

Comment (LiKang6688) [reply .circleci/config.yml:49]: good point.

Comment (LiKang6688) [reply .circleci/config.yml:49]: Thanks. Updated.

Comment (haxxmaxx) [reply .circleci/config.yml]: nice, make sure to do some sanity testing, could have some side-effects

**Review by haxxmaxx — APPROVED** [review]

**Pull request merged**